apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -363,41 +362,36 @@ def meta(self) -> "MetadataWrapper":
     def ndim(self) -> int:
         return len(self._handle.dimension_names)
 
-    def _get_and_cast_domain(
-        self, domain_slot_getter: Callable[[str, DTypeLike], Tuple[object, object]]
+    def _cast_domainish(
+        self, domainish: List[Any]
```

Comment:
Would the type for `domainish` be something like `List[np.array[object, object], ...]`?

Reply:
yeah something like that

it gave me fits & I did `Any` to unblock & forgot to come back & fine-tune ... let me try that ...

thank you! :)

Reply:
@nguyenv I merged without doing this (I resolved the wrong comment) -- will address in follow-up

Reply:
https://github.com/single-cell-data/TileDB-SOMA/issues/3081

Reply:
@nguyenv I may leave it as `Any` but with a comment. Here's what it looks like in one example:

```
>>> sdf._handle._handle.domain()
[<pyarrow.lib.LargeStringArray object at 0x100d8c6a0>
[
  "apple",
  "zebra"
], <pyarrow.lib.Int32Array object at 0x1612e5e40>
[
  -1000000,
  1000000
]]
>>> sdf._handle._handle.domain()[0]
<pyarrow.lib.LargeStringArray object at 0x1612e5ea0>
[
  "apple",
  "zebra"
]
>>> sdf._handle._handle.domain()[1]
<pyarrow.lib.Int32Array object at 0x1612e5e40>
[
  -1000000,
  1000000
]
```

So it's a list of `pyarrow` arrays ...